telethon/tl/session.py

```diff
@@ -64,7 +64,10 @@ def __init__(self, session_user_id):
         self._last_msg_id = 0  # Long
 
         # These values will be saved
-        self.server_address = '91.108.56.165'
+        if use_v6:
+            self.server_address = '[2001:67c:4e8:f002::a]'
```

Comment:
So maybe the problem was the address wasn't enclosed with square braces? Why is this?

Reply:
AFAIK, ipv6 address must be enclosed in square brackets if it used with port or within URL. But I'm not exactly sure, that these brackets are needed also there.